source/custom/xnetsocket_converters.h

```diff
@@ -191,6 +191,56 @@ struct TimeValInputConverter {
   nxtimeval time_val;
 };
 
+struct VirtualInterfaceOutputConverter {
+  VirtualInterfaceOutputConverter() : virtual_interface_ptr(nullptr)
+  {
+  }
+
+  nxVirtualInterface_t** operator&()
+  {
+    return &virtual_interface_ptr;
+  }
+
+  void to_grpc(pb_::RepeatedPtrField<VirtualInterface>& output)
+  {
+    auto curr_vi_ptr = virtual_interface_ptr;
+    for (
+        auto curr_vi_ptr = virtual_interface_ptr;
+        curr_vi_ptr != nullptr;
+        curr_vi_ptr = curr_vi_ptr->nextVirtualInterface) {
+      auto curr_grpc_vi = output.Add();
+      curr_grpc_vi->set_xnet_interface_name(curr_vi_ptr->xnetInterfaceName);
+      curr_grpc_vi->set_mac_address(curr_vi_ptr->macAddress);
+      curr_grpc_vi->set_mac_mtu(curr_vi_ptr->macMTU);
+      curr_grpc_vi->set_operational_status(curr_vi_ptr->operationalStatus);
+      curr_grpc_vi->set_if_index(curr_vi_ptr->ifIndex);
+      for (
+          auto curr_ip_addr_ptr = curr_vi_ptr->firstIPAddress;
+          curr_ip_addr_ptr != nullptr;
+          curr_ip_addr_ptr = curr_ip_addr_ptr->nextIPAddress) {
+        auto curr_grpc_ip_addr = curr_grpc_vi->add_ip_addresses();
+        curr_grpc_ip_addr->set_family(curr_ip_addr_ptr->family);
+        curr_grpc_ip_addr->set_address(curr_ip_addr_ptr->address);
+        curr_grpc_ip_addr->set_net_mask(curr_ip_addr_ptr->netmask);
+        curr_grpc_ip_addr->set_prefix_length(curr_ip_addr_ptr->prefixLength);
+      }
+      for (
+          auto curr_gateway_addr = curr_vi_ptr->firstGatewayAddress;
+          curr_gateway_addr != nullptr;
+          curr_gateway_addr = curr_gateway_addr->nextGatewayAddress) {
+        auto curr_grpc_gateway_addr = curr_grpc_vi->add_gateway_addresses();
+        curr_grpc_gateway_addr->set_family(curr_gateway_addr->family);
+        curr_grpc_gateway_addr->set_address(curr_gateway_addr->address);
+      }
+    }
+    // Free the stack info after we've read it.
+    nxIpStackFreeInfo(virtual_interface_ptr);
```

Comment:
Unlike the other converters, this code is not very testable because I don't have a way to create/free it the way the driver does.

Reply:
How will this get called? I didn't see it referenced anywhere?